### PR TITLE
The last few fixes for MSVC

### PR DESCRIPTION
**Pull request by Orvid**
I didn't want to deal with creating separate PRs for each of these, so this is one still small, PR instead.

And no, I can't get rid of T2 in Future.h, nor can I check T2 against Unit. I don't know why, but MSVC just doesn't like it.

### CODE REVIEW

File: folly/futures/Future.h [hunk 1]
```diff
@@ -62,7 +62,7 @@ class Future {
 
   template <class T2 = T, typename =
             typename std::enable_if<
-              std::is_same<Unit, T2>::value>::type>
+              std::is_same<Unit, T>::value>::type>
```
Comment (JoelMarcey) [inline folly/futures/Future.h]: Why are you doing this?

File: folly/json.cpp [hunk 2]
```diff
@@ -663,7 +663,7 @@ void escapeString(StringPiece input,
         // checking that utf8 encodings are valid
         char32_t v = decodeUtf8(q, e, opts.skip_invalid_utf8);
         if (opts.skip_invalid_utf8 && v == U'\ufffd') {
-          out.append("\ufffd");
+          out.append(u8"\ufffd");
```
Comment (JoelMarcey) [inline folly/json.cpp]: What is this?

**Comment by JoelMarcey** [comment]
A couple of questions from @yfeldblum inline with code

Note: We are going to start our automated pull into external phabricator starting with newer pull requests, so code comments, etc. will be on external diffs instead of me being a conduit.

File: folly/json.cpp (hunk 2 above)
Comment (Orvid) [inline folly/json.cpp]: It's a utf8 string literal, added as part of c++11, and the only way to get msvc to allow codepoints over ff in a narrow string literal.

File: folly/futures/Future.h (hunk 1 above)
Comment (Orvid) [inline folly/futures/Future.h]: Because MSVC is dumb and this condition doesn't work if it's T2.

**Comment by Orvid** [comment]
Yep, I saw the diffs get created in phabricator a while ago. I assume the links to them will be added to the PRs once the pull os complete?

**Comment by JoelMarcey** [comment]
@Orvid For ones that I have already pulled manually internally, we are just going to use the *old* process. For pull requests above #309, we will start using the actually phabricator importer, etc.

**Comment by Orvid** [comment]
Moved internally to D2901301.

**Pull request closed without merging**